components/header-bar/src/online-status.js

```diff
@@ -1,15 +1,35 @@
-import { useOnlineStatus } from '@dhis2/app-runtime'
+import { useConfig, useOnlineStatus } from '@dhis2/app-runtime'
 import cx from 'classnames'
+import moment from 'moment'
 import PropTypes from 'prop-types'
 import React from 'react'
 import i18n from './locales/index.js'
 import styles from './online-status.styles.js'
 
 /** A badge to display online/offline status in the header bar */
-export function OnlineStatus({ info, dense }) {
-    const { online } = useOnlineStatus()
+export function OnlineStatus({ dense }) {
+    const { online, lastOnline } = useOnlineStatus()
+    const { headerbar } = useConfig()
+
     const displayStatus = online ? i18n.t('Online') : i18n.t('Offline')
 
+    let info
```

Comment:
Extracting this logic into a hook is a bit cleaner as there's no need for mutable variables:

```js
const useOnlineStatusInfo = ({ online, lastOnline }) => {
    const { headerbar } = useConfig()
    switch (headerbar?.onlineStatusInfo) {
        case 'last-online': 
            if (!online && lastOnline) {
                 return i18n.t('Last online {{relativeTime}}', {
                     relativeTime: moment(lastOnline).fromNow(),
                 })
            } else {
                return null
            }
        default:
            return null
    }
    return null
}
```

```js
const info = useOnlineStatusInfo()
```

As there's currently only one supported value for `headerbar.onlineStatusInfo`, I'd even be inclined to do:

```js
const useOnlineStatusInfo = ({ online, lastOnline }) => {
    const { headerbar } = useConfig()
    if (headerbar?.onlineStatusInfo === 'last-online' && !online && lastOnline) {
        return i18n.t('Last online {{relativeTime}}', {
            relativeTime: moment(lastOnline).fromNow(),
        })
    }
    return null
}
```

Reply:
That looks clean 👍